sympy/physics/continuum_mechanics/cross_section.py

```diff
@@ -0,0 +1,208 @@
+from sympy.printing import sstr
+from sympy.geometry import Ellipse, Polygon
+
+
+class CrossSection(object):
+    """
+    A class to define the cross-section of any structural element,
+    especially beam.
+    The CrossSection class supports any arbitrary geometric shape to be defined
+    as the cross-section of the structure.
+    It gives functionality to provide different properties of the cross-section.
+
+    Attributes
+    ==========
+
+    geometry: an object of the geometry module
+
+    Examples
+    ========
+
+    >>> from sympy import Polygon
+    >>> from sympy.physics.continuum_mechanics.cross_section import CrossSection
+    >>> p = CrossSection(Polygon((0, 0), (9, 0), (9, 13), (0, 13)))
+    >>> p.shape
+    Polygon(Point2D(0, 0), Point2D(9, 0), Point2D(9, 13), Point2D(0, 13))
+    >>> p.centroid()
+    Point2D(9/2, 13/2)
+    >>> p.area()
+    117
+    >>> p.second_moment()
+    (6591/4, 3159/4, 0)
+    >>> p.polar_modulus()
+    4875/2
+    >>> p.section_modulus()
+    (507/2, 351/2)
+
+    >>> from sympy import Circle
+    >>> c = CrossSection(Circle((15, 15), 15))
+    >>> c.centroid()
+    Point2D(15, 15)
+    >>> c.area()
+    225*pi
+    >>> c.second_moment()
+    (50625*pi/4, 50625*pi/4, 0)
+    >>> c.polar_modulus()
+    50625*pi/2
+    >>> c.section_modulus()
+    (3375*pi/4, 3375*pi/4)
+    """
+    def __init__(self, shape):
+        self.shape = shape
+
+    def __str__(self):
+        str_1 = 'CrossSection({})'.format(sstr(self._shape))
+        return str_1
+
+    __repr__ = __str__
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @shape.setter
+    def shape(self, shape):
+        self._shape = shape
+
+    def area(self):
+        """Returns the area of the cross-section geometry"""
+        return self.shape.area
+
+    def centroid(self):
+        """Returns a point with the centroid coordinates"""
+        if isinstance(self.shape, Ellipse):
+            return self.shape.center
+        return self.shape.centroid
```

Comment:
It would be better to add a centroid property to Ellipse rather than special-casing it here